tests/unit/lms/views/api/canvas/sync_test.py

```diff
@@ -1,140 +1,148 @@
-    ]
     canvas_api_client.course_sections.assert_called_once_with(
-        "test_custom_canvas_course_id"
+        request_json["course"]["custom_canvas_course_id"]
     )
-    assert_that_it_called_sync_and_returned_the_groupids(
-        lti_h_service, returned_groupids, expected_groups
+    assert_sync_and_return(
+        groupids, synced_sections=sections.course, returned_sections=sections.course,
     )
 
 
-def test_sync_when_in_SpeedGrader(pyramid_request, canvas_api_client, lti_h_service):
-    pyramid_request.lti_user = pyramid_request.lti_user._replace(roles="Instructor")
-    pyramid_request.json["learner"] = {
-        "canvas_user_id": "test_canvas_user_id",
-    }
+@pytest.mark.usefixtures("is_instructor")
+@pytest.mark.usefixtures("is_speedgrader")
+def test_sync_when_in_SpeedGrader(
+    pyramid_request, canvas_api_client, sections, assert_sync_and_return, request_json
+):
+    groupids = sync(pyramid_request)
 
-    returned_groupids = sync(pyramid_request)
-
-    instructors_groups = [
-        factories.HGroup(
-            name="Section 1",
-            authority_provided_id="d0f36006728f2277f228b74c8a7f620305bfb3e7",
-            type="section_group",
-        ),
-        factories.HGroup(
-            name="Section 2",
-            authority_provided_id="d99674b9700f4a40a2b301d2949b61339c58236c",
-            type="section_group",
-        ),
-        factories.HGroup(
-            name="Section 3",
-            authority_provided_id="6a85e3651705dee4da0805b8985472343cbea94e",
-            type="section_group",
-        ),
-    ]
-    canvas_api_client.course_sections.assert_called_once_with(
-        "test_custom_canvas_course_id"
-    )
-    lti_h_service.sync.assert_called_once_with(instructors_groups, {"foo": "bar"})
-    canvas_api_client.users_sections.assert_called_once_with(
-        "test_canvas_user_id", "test_custom_canvas_course_id"
+    course_id = request_json["course"]["custom_canvas_course_id"]
+    user_id = request_json["learner"]["canvas_user_id"]
+    canvas_api_client.course_sections.assert_called_once_with(course_id)
+    canvas_api_client.users_sections.assert_called_once_with(user_id, course_id)
+
+    assert_sync_and_return(
+        groupids, synced_sections=sections.course, returned_sections=sections.user,
     )
-    assert returned_groupids == [
-        "group:d99674b9700f4a40a2b301d2949b61339c58236c@TEST_AUTHORITY"
-    ]
 
 
-def assert_that_it_called_sync_and_returned_the_groupids(
-    lti_h_service, returned_groupids, expected_groups
-):
-    lti_h_service.sync.assert_called_once_with(expected_groups, {"foo": "bar"})
-    assert returned_groupids == [
-        f"group:{group.authority_provided_id}@TEST_AUTHORITY"
-        for group in expected_groups
-    ]
+pytestmark = pytest.mark.usefixtures("canvas_api_client", "lti_h_service")
 
 
-pytestmark = pytest.mark.usefixtures("canvas_api_client", "lti_h_service")
+@pytest.fixture
+def assert_sync_and_return(lti_h_service, request_json):
```

Comment:
This is the main guy that pulls all of this together. It roughly maps to the previous `assert_that_it_called_sync_and_returned_the_groupids` but with the following changes:

 * It accepts sections and converts them into the right groups itself
 * It uses groupid function instead of replicating the logic
 * It accepts two lists of sections, instead of one, allowing it to be used in all three tests (the speedgrader one has different values)